MetadataExtractor/Util/FileTypeDetector.cs

```diff
@@ -94,7 +103,7 @@ public static FileType DetectFileType(Stream stream)
             if (!stream.CanSeek)
                 throw new ArgumentException("Must support seek", nameof(stream));
 
-            var maxByteCount = _root.MaxDepth;
+            var maxByteCount = Math.Max(_root.MaxDepth, TgaHeaderReader.HeaderSize);
```

Comment:
We should change `_fixedCheckers` to be an enumerable of value tuples with an additional `int? HeaderSize`, then scan all checkers for their max non-null size for use here. It'll force us to think about this requirement for the other existing checkers and future ones too.

Reply:
Maybe extract an interface so the check method and the size value are closer together?

Reply:
Interface works.

Reply:
Done.